apps/src/templates/projects/ProjectCardGrid.jsx

```diff
@@ -32,28 +40,21 @@ const ProjectCardGrid = React.createClass({
     }).isRequired,
     galleryType: PropTypes.oneOf(['personal', 'class', 'public']).isRequired,
     selectedGallery: PropTypes.string.isRequired
-  },
-
-  getInitialState() {
-    return {
-      showAll: true,
-      showApp: ''
-    };
-  },
+  };
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.selectedGallery !== this.props.selectedGallery && nextProps.selectedGallery === Galleries.PUBLIC) {
       this.setState({showAll: true, showApp: ''});
     }
-  },
+  }
 
   onSelectApp(appType) {
     this.setState({showAll: false, showApp: appType});
-  },
+  }
 
   viewAllProjects() {
     this.setState({showAll: true, showApp: ''});
-  },
+  }
```

Comment:
`onSelectApp` and `viewAllProjects` do not appear to be properly bound to `this` here. I believe the preferred syntax is:
```javascript
  onSelectApp = appType => {
    this.setState({showAll: false, showApp: appType});
  }
```
@islemaster has done a bunch of these conversions and may be able to offer a second opinion. Here is one very long explanation, or you can skip to the end: http://reactkungfu.com/2015/07/why-and-how-to-bind-methods-in-your-react-component-classes/

Reply:
Right, good catch!  I will fix.